core/src/test/java/io/questdb/std/fastdouble/AbstractDoubleHandPickedTest.java

```diff
@@ -0,0 +1,340 @@
+                        Double.NEGATIVE_INFINITY)),
+                dynamicTest(Double.toHexString(Double.NaN), () -> testLegalHexInput(
+                        Double.NaN)),
+                dynamicTest(Double.toString(Math.nextUp(0.0f)), () -> testLegalHexInput(
+                        Math.nextUp(0.0))),
+                dynamicTest(Double.toString(Math.nextDown(0.0f)), () -> testLegalHexInput(
+                        Math.nextDown(0.0))),
+                dynamicTest(Double.toHexString(Math.nextUp(0.0)), () -> testLegalHexInput(
+                        Math.nextUp(0.0))),
+                dynamicTest(Double.toHexString(Math.nextDown(0.0)), () -> testLegalHexInput(
+                        Math.nextDown(0.0))),
+                dynamicTest("Just above MAX_VALUE: 0x1.fffffffffffff8p1023", () -> testLegalInput(
+                        "0x1.fffffffffffff8p1023", Double.POSITIVE_INFINITY)),
+                dynamicTest("Just below MIN_VALUE: 0x0.00000000000008p-1022", () -> testLegalInput(
+                        "0x0.00000000000008p-1022", 0.0))
+        );
+    }
+
+    @TestFactory
+    List<DynamicNode> dynamicTestsLegalInputsWithPrefixAndSuffix() {
+        return Arrays.asList(
+                dynamicTest("before-1after", () -> testLegalInputWithPrefixAndSuffix("before-1after", 6, 2, -1.0)),
+                dynamicTest("before7.789after", () -> testLegalInputWithPrefixAndSuffix("before7.789after", 6, 5, 7.789)),
+                dynamicTest("before7.78e2after", () -> testLegalInputWithPrefixAndSuffix("before7.78e2after", 6, 6, 7.78e2)),
+                dynamicTest("before0x123.4p0after", () -> testLegalInputWithPrefixAndSuffix("before0x1234p0after", 6, 8, 0x1234p0)),
+                dynamicTest("before0x123.45p0after", () -> testLegalInputWithPrefixAndSuffix("before0x123.45p0after", 6, 10, 0x123.45p0)),
+                dynamicTest("Outside Clinger fast path (min_clinger_significand + 1, min_clinger_exponent - 1)", () -> testLegalInputWithPrefixAndSuffix(
+                        "before1e-23after", 6, 5, 1e-23)),
+                dynamicTest("before9007199254740992.e-256after", () -> testLegalInputWithPrefixAndSuffix(
+                        "before9007199254740992.e-256after", 6, 22, 9007199254740992.e-256))
+
+
+        );
+    }
+
+
+    @TestFactory
+    Stream<DynamicNode> dynamicTestsPowerOfTen() {
+        return IntStream.range(-307, 309).mapToObj(i -> "1e" + i)
+                .map(d -> dynamicTest(d, () -> testLegalInput(d, Double.parseDouble(d))));
+    }
+
+    abstract double parse(CharSequence str) throws NumericException;
+
+    protected abstract double parse(String str, int offset, int length) throws NumericException;
+
+    @TestFactory
+    Stream<DynamicNode> testErrorCases() throws IOException, URISyntaxException {
+        Path p = Paths.get(AbstractDoubleHandPickedTest.class.getResource("/fastdouble/FastDoubleParser_errorcases.txt").toURI());
```

Comment:
nit: the file seems to contain non-error cases while test and file names suggest otherwise.

Reply:
I kept the original file name. Now i'm struggling to come up with file name. Content seems to be mixed, for example:

```
925479147622538608157385395783430387663692794943115504935349235893227159976959228515625
929769423252076208907803604252402073697828855693148231154054883518256247043609619140625
9.2997877719475087645901797e+86
```

Reply:
How about `FastDoubleParser_testcases.txt`?

Reply:
done in https://github.com/questdb/questdb/pull/2527/commits/078097ddfa228a49572c90317d43c51035525935